core/src/rpc.rs

```diff
@@ -1397,20 +1487,32 @@ fn get_encoded_account(
     Ok(response)
 }
 
-/// Use a set of filters to get an iterator of keyed program accounts from a bank
-fn get_filtered_program_accounts(
-    bank: &Arc<Bank>,
-    program_id: &Pubkey,
-    filters: Vec<RpcFilterType>,
-) -> impl Iterator<Item = (Pubkey, Account)> {
-    bank.get_program_accounts(&program_id)
-        .into_iter()
-        .filter(move |(_, account)| {
-            filters.iter().all(|filter_type| match filter_type {
-                RpcFilterType::DataSize(size) => account.data.len() as u64 == *size,
-                RpcFilterType::Memcmp(compare) => compare.bytes_match(&account.data),
-            })
-        })
+fn get_spl_token_owner_filter(program_id: &Pubkey, filters: &[RpcFilterType]) -> Option<Pubkey> {
+    if program_id != &spl_token_id_v2_0() {
+        return None;
+    }
+    let mut data_size_filter: Option<u64> = None;
+    let mut owner_key: Option<Pubkey> = None;
+    for filter in filters {
+        match filter {
+            RpcFilterType::DataSize(size) => data_size_filter = Some(*size),
+            RpcFilterType::Memcmp(Memcmp {
+                offset: PUBKEY_BYTES,
```

Comment:
maybe related to https://github.com/solana-labs/solana/pull/14212/files#r548046729,

this is rather SPL_TOKEN_ACCOUNT_OWNER_OFFSET or something? I mean, this is depending on particular data layout of spl-token account, not the size of Pubkey.

Reply:
Done in https://github.com/solana-labs/solana/pull/14212/commits/d607d1aa5da54b078f1eb27bcfe1752df515072e